src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -135,6 +136,9 @@ public DlsFlsValveImpl(
      */
     @Override
     public boolean invoke(PrivilegesEvaluationContext context, final ActionListener<?> listener) {
+        if (HeaderHelper.isInternalOrPluginRequest(threadContext)) {
+            return true;
+        }
```

Comment:
I missed the previous PR; hence, checking here

We have ensured normal/admin users of the cluster cannot create a user with prefix ```plugin:```, right?

Reply:
`:` is [restricted from usernames](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/user/UserService.java#L92-L94) since its a delimeter used in basic auth. The basic auth header uses `:` as the separator for `{username}:{password}` and Base64 encodes it to create the basic auth token that's send in an HTTP Basic auth request.

The concept of this plugin user only exists in memory and takes advantage of the fact that `:` is forbidden from usernames for security. I thought about including a different attribute on the User object (i.e. type), but think thought this was a reasonable solution that didn't involve any changes to serialization/deserialization for the user object.

Reply:
The alternatives would be to consider something like this [PR](https://github.com/opensearch-project/security/pull/2848). For extensions, https://github.com/opensearch-project/security/pull/2594 introduced a special attribute called `service` that would be an [entry in the attribute map](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/user/User.java#L290-L298), but I don't think that pattern should be copied since the attribute map is modifiable by a cluster admin.

Reply:
thanks, to be safer - should we restrict ``plugin`` as a restricted keyword/user?

Reply:
anyway, this discussion can be outside this PR, approving this one.